arvo/ames/ames.md

```diff
@@ -319,34 +334,45 @@ with `%boon`s and create congestion there. So naxplanations have their own bone,
 and so do acks to packets that make up a naxplanation, as well as the
 message-level naxplanation ack.
 
-### Packet relaying
+### Packet relaying and peer discovery
 
-Here we describes how the Ames network relays packets. We ignore all details
-about UDP, which occurs at a lower layer than Ames.
+Here we describes how the Ames network relays packets and does peer discovery.
+We ignore all details about UDP, which occurs at a lower layer than Ames.
 
 When a ship first contacts another ship, it may only know its `@p`, but an IP
 address and port are necessary for peer-to-peer communication to occur.
 Thus the initial correspondence between two ships, or additional correspondence
 following a change in IP or port of one of the parties, will need to be relayed
 by an intermediary that knows the IP and port of the receiving ship.
 
-For now packet relaying is handled entirely by galaxies. Every galaxy is
+For now, peer discovery is handled entirely by galaxies. Every galaxy is
 responsible for knowing the IP and port of every planet underneath it. In the
-future, galaxies will only need to know the star sponsoring a planet, and stars
-will be responsible for the final step of the relay.
+future, galaxies will only need to know the IP and port of the star sponsoring a
+planet, and stars will be responsible knowing the IP and port of their sponsored planets.
+
+Galaxies are also utilized for packet relaying in the case where two ships
+cannot communicate directly with one another, as is often the case when one or
+both are behind a
+[NAT](https://en.wikipedia.org/wiki/Network_address_translation) or certain
+firewalls. Again, stars will someday assist with this process as well in a
+similar fashion to peer discovery.
+
+In the case of moons, their parent is responsible for packet relaying and peer
```

Comment:
This is a nitpick, but I'd say `their parent planet`, in case the reader forgot the names for different kinds of nodes.

Reply:
For cleaner grammar, I'd use singular 'moon' to match the singular 'parent':  `In the case of a moon, its parent is responsible` ...

Reply:
Since galaxies and stars can have moons, I didn't want to say planet. I changed it to this though:

```
In the case of a moon, its parent ship is responsible for packet relaying and peer
discovery, analagous to the role that galaxies currently play for stars and planets.
```